pysqldb3/tests/test_backup_table.py

```diff
@@ -0,0 +1,523 @@
+import configparser
+import os
+
+from .. import pysqldb3 as pysqldb
+from ..data_io import *
+from .. sql import GET_MS_INDEX_QUERY
+
+config = configparser.ConfigParser()
+config.read(os.path.dirname(os.path.abspath(__file__)) + "\\db_config.cfg")
+
+test_data_dir = os.path.dirname(os.path.abspath(__file__))+'/test_data'
+
+
+db = pysqldb.DbConnect(type=config.get('PG_DB', 'TYPE'),
+                       server=config.get('PG_DB', 'SERVER'),
+                       database=config.get('PG_DB', 'DB_NAME'),
+                       user=config.get('PG_DB', 'DB_USER'),
+                       password=config.get('PG_DB', 'DB_PASSWORD'),
+                       allow_temp_tables=True)
+
+sql = pysqldb.DbConnect(type=config.get('SQL_DB', 'TYPE'),
+                        server=config.get('SQL_DB', 'SERVER'),
+                        database=config.get('SQL_DB', 'DB_NAME'),
+                        user=config.get('SQL_DB', 'DB_USER'),
+                        password=config.get('SQL_DB', 'DB_PASSWORD'),
+                        allow_temp_tables=True)
+
+test_pg_to_backup = 'test_pg_to_backup_{}'.format(db.user)
+test_pg_from_backup = 'test_pg_from_backup_{}'.format(db.user)
+test_sql_to_backup = 'test_sql_to_backup_{}'.format(db.user)
+test_sql_from_backup = 'test_sql_from_backup_{}'.format(db.user)
+
+ms_schema = 'dbo'
+pg_schema = 'working'
+
+test_back_file = test_data_dir+'/backup.sql'
+
+
+# org_table = 'daylighting_old_turn_calming_20240626'
+# org_schema= 'minireports'
+
+class TestBackupTablesPg:
+    def test_backup_tables_basic(self):
+        db.drop_table(table=test_pg_to_backup, schema=pg_schema)
+
+        # table schema
+        db.query(f"""
+            CREATE TABLE {pg_schema}.{test_pg_to_backup} (
+                id int,
+                column2 text,
+                column3 timestamp,
+                "1 test messy column" text
+            );
+        """)
+        # populate table
+        for i in range(10):
+            db.query(f"""
+                INSERT INTO {pg_schema}.{test_pg_to_backup}
+                    (id, column2, column3, "1 test messy column")
+                values ({i}, '{i}', '{'2022-10-14 10:12:40-04'}', '{'test '*i}')
+            """)
+        # validate table created
+        assert db.table_exists(test_pg_to_backup, schema=pg_schema)
+        db.query(f"select count(*) cnt from {pg_schema}.{test_pg_to_backup}")
+        assert db.data[0][0] == 10
+
+        if os.path.isfile(test_back_file):
+            os.remove(test_back_file)
+        assert not os.path.isfile(test_back_file)
+
+        db.backup_table(pg_schema, test_pg_to_backup, test_back_file, pg_schema, test_pg_from_backup)
+        assert os.path.isfile(test_back_file)
+
+        # run backup
+        db.drop_table(pg_schema, test_pg_from_backup)
+        schema_table_name = db.create_table_from_backup(test_back_file)
```

Comment:
I fixed this in my branch for the capitalized table names (so no need to fix it in this version, but I'm letting you know), but I wanted to point out that the `schema_table_name` variable is never referenced again and doesn't appear in any other test, so I removed the variable creation.

Reply:
added a check that its returning correctly